test/test.ml

```diff
@@ -396,32 +391,35 @@ module Make(IO : Redis.S.IO) = struct
       (fun scanned_fields ->
          List.length fields = List.length scanned_fields)
 
+  let cleanup_keys conn =
+    Client.keys conn "ounit_*" >>= Client.del conn
 
   let bracket test_case () =
-    IO.run
-      (setup () >>= fun conn ->
-       test_case conn >>= fun () ->
-       teardown conn)
+    IO.run @@ Client.with_connection redis_spec test_case
```

Comment:
I didn't expect that... but it should be helpful none the less, particularly when this makes it into opam.